cognite/client/data_classes/hosted_extractors/sources.py

```diff
@@ -577,6 +578,366 @@ class MQTT5SourceUpdate(_MQTTUpdate):
+        return output
+
+
+class KafkaSourceUpdate(SourceUpdate):
+    _type = "kafka"
+
+    class _BootstrapBrokersUpdate(CognitePrimitiveUpdate):
+        def set(self, value: Sequence[KafkaBroker]) -> KafkaSourceUpdate:
+            return self._set([broker.dump() for broker in value])
+
+    class _AuthenticationUpdate(CognitePrimitiveUpdate):
+        def set(self, value: MQTTAuthentication | None) -> KafkaSourceUpdate:
+            return self._set(value.dump() if value else None)
+
+    class _UseTlsUpdate(CognitePrimitiveUpdate):
+        def set(self, value: bool) -> KafkaSourceUpdate:
+            return self._set(value)
+
+    class _CACertificateUpdate(CognitePrimitiveUpdate):
+        def set(self, value: CACertificate | None) -> KafkaSourceUpdate:
+            return self._set(value.dump() if value else None)
+
+    class _AuthCertificateUpdate(CognitePrimitiveUpdate):
+        def set(self, value: AuthCertificate | None) -> KafkaSourceUpdate:
+            return self._set(value.dump() if value else None)
+
+    @property
+    def bootstrap_brokers(self) -> _BootstrapBrokersUpdate:
+        return KafkaSourceUpdate._BootstrapBrokersUpdate(self, "bootstrapBrokers")
+
+    @property
+    def authentication(self) -> _AuthenticationUpdate:
+        return KafkaSourceUpdate._AuthenticationUpdate(self, "authentication")
+
+    @property
+    def useTls(self) -> _UseTlsUpdate:
+        return KafkaSourceUpdate._UseTlsUpdate(self, "useTls")
+
+    @property
+    def ca_certificate(self) -> _CACertificateUpdate:
+        return KafkaSourceUpdate._CACertificateUpdate(self, "caCertificate")
+
+    @property
+    def auth_certificate(self) -> _AuthCertificateUpdate:
+        return KafkaSourceUpdate._AuthCertificateUpdate(self, "authCertificate")
+
+    @classmethod
+    def _get_update_properties(cls, item: CogniteResource | None = None) -> list[PropertySpec]:
+        return [
+            PropertySpec("bootstrap_brokers", is_nullable=False),
+            PropertySpec("authentication", is_nullable=True, is_object=True),
+            PropertySpec("useTls", is_nullable=False),
+            PropertySpec("ca_certificate", is_nullable=True, is_object=True),
+            PropertySpec("auth_certificate", is_nullable=True, is_object=True),
+        ]
+
+
+class RestSourceWrite(SourceWrite):
+    """A hosted extractor source represents an external source system on the internet.
+    The source resource in CDF contains all the information the extractor needs to
+    connect to the external source system.
+
+    This is the write/request format of the rest resource.
+
+    Args:
+        external_id (str): The external ID provided by the client. Must be unique for the resource type.
+        host (str): Host or IP address to connect to.
+        scheme (Literal["http", "https"]): Type of connection to establish.
+        port (int): Port on server to connect to. Uses default ports based on the scheme if omitted.
```

Comment:
This comment indicates that it can be omitted, but __init__ forces a value to be given.